src/regressionmodel.jl

```diff
@@ -117,3 +117,27 @@ Return the offset used in the model, i.e. the term added to the linear predictor
 known coefficient 1.
 """
 function offset end
+
+"""
+    linearpredictor(model::RegressionModel)
+
+Return the model's linear predictor, `Xβ` where `X` is the model matrix and `β` is the
+vector of coefficients, or `Xβ + offset` if the model was fit with an offset.
+"""
+function linearpredictor(model::RegressionModel)
+    X = modelmatrix(model)
+    β = coef(model)
+    η = X * β
+    off = offset(model)
+    if off !== nothing && !isempty(off)
```

Comment:
Nothing in the current docs mentions what a model should return if there's no offset. Allowing to return an empty vector doesn't sound like a good idea, as `nothing` is more appropriate. Anyway it would be good to add this to the docstring for `offset`.

Reply:
I agree that `nothing` is more appropriate, though an empty vector is the convention used by GLM, which is why I accounted for it here.

Reply:
OK. Probably better change GLM (or add a special method there if we don't want to break the API right now) then.

Reply:
Ah, GLM doesn't actually extend this function (yet) so we're fine. I've added a note to the `offset` docstring as a separate commit.